File: src/test/java/net/tassia/util/Base16Test.java
package net.tassia.util;

import net.tassia.Tests;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

public class Base16Test {

	@Test
	public void testRandomized() {
		for (int i = 0; i < Tests.ENTROPY_ITERATIONS; i++) {
			byte[] data = Tests.randomData();

			String encoded = Base16.ENCODER.encode(data);
			byte[] decoded = Base16.DECODER.decode(encoded);

			Assertions.assertArrayEquals(data, decoded);
		}
	}

	@Test
	public void testReferences() {
		// TODO
	}

}

File: commons-encoding/src/test/kotlin/net/tassia/encoding/Base16Test.kt
package net.tassia.encoding

import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.DisplayName
import org.junit.jupiter.api.Test
import kotlin.random.Random

@DisplayName("Base16")
class Base16Test {

	@Test
	@DisplayName("Randomized")
	fun testRandomized() {
		repeat(4096 * 8) {

			// Generate random data
			val data = Random.nextBytes(Random.nextInt(4096))

			// Encode & decode
			val encoded = data.base16()
			val decoded = encoded.base16()

			// Assert
			Assertions.assertTrue(data contentEquals decoded)

		}
	}

}
